src/chainstate/burn/distribution.rs

```diff
@@ -38,195 +39,301 @@ use util::log;
+                    //  user burn op in the burn samples
+                    user_burn.burn_fee = per_recipient;
+
+                    for recipient in user_burn_recipients.iter() {
+                        let recipient_commit = commits_with_priors[*recipient]
+                            .get_mut(window_size as usize - 1 - rel_block_height)
+                            .expect("BUG: (window_size - i) should be in window range")
+                            .as_mut()
+                            .expect("BUG: Should have a non-none commit entry");
+                        // cheap sanity checks
+                        assert_eq!(
+                            recipient_commit.op.key_block_ptr,
+                            user_burn_target_key.key_block_ptr
+                        );
+                        assert_eq!(
+                            recipient_commit.op.key_vtxindex,
+                            user_burn_target_key.key_vtxindex
+                        );
+                        // are we at the last block in the window?
+                        //  if so, track the user burn op
+                        if rel_block_height as u8 == window_size - 1 {
+                            if let Some(user_burns) =
+                                commit_txid_to_user_burns.get_mut(&recipient_commit.op.txid)
+                            {
+                                user_burns.push(user_burn.clone());
+                            } else {
+                                commit_txid_to_user_burns.insert(
+                                    recipient_commit.op.txid.clone(),
+                                    vec![user_burn.clone()],
+                                );
+                            }
+                        }
+
+                        recipient_commit.user_burns += per_recipient;
+                    }
                 }
             }
+        }
 
-            for i in 0..user_burns.len() - 1 {
-                if user_burns[i].vtxindex >= user_burns[i + 1].vtxindex {
-                    panic!("FATAL ERROR: user burns are not in order");
+        // now, commits_with_priors has the burn amounts and user burn supports for each
+        //   linked commitment, we can now generate the burn sample points.
+        let mut burn_sample = commits_with_priors
+            .into_iter()
+            .map(|mut linked_commits| {
+                let mut all_burns: Vec<_> = linked_commits
+                    .iter()
+                    .map(|commit| {
+                        if let Some(commit) = commit {
+                            (commit.op.burn_fee as u128) + (commit.user_burns as u128)
+                        } else {
+                            // use 1 as the linked commit min. this gives a miner a _small_
+                            //  chance of winning a block even if they haven't performed chained utxos yet
+                            1
```

Comment:
What happens if this is 0?

Reply:
It leads to a corner case where if there's no mining for some period of time, no miner will get pick for 3-5 blocks once mining resumes.